core/src/apps/common/request_pin.py

```diff
@@ -124,18 +124,28 @@ async def verify_user_pin(
 
 
 async def error_pin_invalid(ctx: wire.Context) -> NoReturn:
-    from apps.common.confirm import confirm
-
-    text = Text("Wrong PIN", ui.ICON_WRONG, ui.RED)
-    text.normal("The PIN you entered is", "invalid.")
-    await confirm(ctx, text, confirm=None, cancel="Close")
+    from trezor.ui.layouts import show_error_and_raise
+
+    await show_error_and_raise(
+        ctx,
+        "warning_wrong_pin",
+        header="Wrong PIN",
+        content="The PIN you entered is invalid.",
+        red=True,
+        exc=wire.PinInvalid,
+    )
     raise wire.PinInvalid
```

Comment:
isn't this dead code now?

Reply:
It is but mypy needs it:
```
src/apps/common/request_pin.py:140: error: Implicit return in function which does not return  [misc]
```
Which is weird because `show_error_and_raise` is declared as `NoReturn`.

Reply:
that's what I was thinking too. weird indeed. maybe something to do with asyncness of the function?
in any case let's put in an `assert False` instead to make it clearer for the reader

Reply:
Fixed in fef81373696d5dc2929b40dba0343afe855c9a7f. `assert False` does nothing in production firmware, right?

Reply:
correct